cirq-core/cirq/work/observable_measurement.py

```diff
@@ -530,7 +542,7 @@ def measure_grouped_settings(
     for max_setting, param_resolver in itertools.product(
         grouped_settings.keys(), study.to_resolvers(circuit_sweep)
     ):
-        circuit_params = param_resolver.param_dict
+        circuit_params = cast(Dict[str, float], param_resolver.param_dict)
```

Comment:
I think we would probably need to change the type accepted by `_MeasurementSpec` so it accepts a dictionary from `Dict[str, Union[float, complex])`; but we can add a #type: ignore here and open another issue to track this change. I don't want to block this PR for this change.

Reply:
Added #5282 to track